dashboard/frontend/main.js

```diff
@@ -0,0 +1,76 @@
+const socket = io(undefined, {
+    auth: {
+        id: 'gxwYN8wa0kerK20F',
+        type: 'frontend',
+    },
+});
+
+const execRuns = document.getElementById('exec-runs');
+
+socket.on('connect', () => {
+    console.log('connected to broker.');
+});
+
+// on clicking the pause button
+const clickPauseButton = (value) => {
+    // get process id of the process on which pause button has been clicked
+    const processId = value.parentElement.getAttribute('data-process-id');
+
+    // if the run has not ended
+    if (expiredRuns.indexOf(processId) === -1) {
+        const currVal = document.getElementById(
+            `pauseButton-${processId}`
+        ).innerText;
```

Comment:
This seems hacky and has potential to break easily. Instead, store the state of the run and take decision based on that. Don't use UI as the store.